Jellyfin.Api/Controllers/DynamicHlsController.cs

```diff
@@ -1415,14 +1486,25 @@ private string GetAudioArguments(StreamState state, EncodingOptions encodingOpti
 
             if (EncodingHelper.IsCopyCodec(audioCodec))
             {
-                var videoCodec = _encodingHelper.GetVideoEncoder(state, encodingOptions);
+                var videoCodec = _encodingHelper.GetVideoEncoder(state, _encodingOptions);
+                var segmentFormat = HlsHelpers.GetSegmentFileExtension(state.Request.SegmentContainer).TrimStart('.');
+                var bitStreamArgs = string.Empty;
+
+                // Apply aac_adtstoasc bitstream filter when media source is in mpegts.
+                if (string.Equals(segmentFormat, "mp4", StringComparison.OrdinalIgnoreCase)
+                    && (string.Equals(state.MediaSource.Container, "mpegts", StringComparison.OrdinalIgnoreCase)
+                        || string.Equals(state.MediaSource.Container, "hls", StringComparison.OrdinalIgnoreCase)))
+                {
+                    bitStreamArgs = _encodingHelper.GetBitStreamArgs(state.AudioStream);
+                    bitStreamArgs = string.IsNullOrEmpty(bitStreamArgs) ? string.Empty : " " + bitStreamArgs;
+                }
```

Comment:
This code is duplicated from above? Maybe extract it to a private function `GetAudioBitStreamArguments`?